enhancements/installer/vsphere-ipi-multiple-datacenter-clusters.md

```diff
@@ -0,0 +1,1169 @@
+  // server is the fully-qualified domain name or the IP address of the vCenter server.
+  // +kubebuilder:validation:Required
+  // +kubebuilder:validation:MaxLength=255
+  Server string `json:"server"`
+
+  // port is the TCP port that will be used to communicate to
+  // the vCenter endpoint. This is typically unchanged from
+  // the default of HTTPS TCP/443.
+  // +kubebuilder:validation:Optional
+  // +kubebuilder:validation:Minimum=1
+  // +kubebuilder:validation:Maximum=32767
+  // +kubebuilder:default=443
+  Port uint `json:"port,omitempty"`
+
+  // insecureFlag is a boolean, if set to true
+  // will disable certificate verification and
+  // set Go's `tls.Config.InsecureSkipVerify` to true.
+  // > If `InsecureSkipVerify` is true, TLS accepts any certificate
+  // > presented by the server and any host name in that certificate.
+  // Setting this value to false will require a valid certificate
+  // or the self-signed to be added to the certificate store.
+  // +kubebuilder:validation:Optional
+  InsecureFlag bool `json:"insecureFlag,omitempty"`
+
+  // Datacenter in which VMs are located.
+  // +kubebuilder:validation:Required
+  // +kubebuilder:validation:MinItems=1
+  Datacenters []string `json:"datacenters"`
+
+  // Soap round tripper count (retries = RoundTripper - 1)
+  // +kubebuilder:validation:Optional
+  RoundTripperCount uint `json:"soapRoundtripCount,omitempty"`
+
+  // Specifies the path to a CA certificate in PEM format. Optional; if not
+  // configured, the system's CA certificates will be used.
+  // +kubebuilder:validation:Optional
+  CAFile string `json:"caFile,omitempty"`
+
+  // Thumbprint of the VCenter's certificate thumbprint
+  // +kubebuilder:validation:Optional
+  Thumbprint string `json:"thumbprint,omitempty"`
+
+  // Name of the secret where vCenter credentials are present.
+  // +kubebuilder:validation:MinLength=1
+  // +kubebuilder:validation:MaxLength=256
+  // +kubebuilder:validation:Required
+  SecretName string `json:"secretName"`
+
+  // Namespace where the secret will be present containing vCenter credentials.
+  // +kubebuilder:validation:Pattern=^[a-z0-9]([-a-z0-9]*[a-z0-9])?$
+  // +kubebuilder:validation:MinLength=1
+  // +kubebuilder:validation:MaxLength=63
+  // +kubebuilder:validation:Required
+  SecretNamespace string `json:"secretNamespace"`
+
+  // IP Family enables the ability to support IPv4 or IPv6
+  // Supported values are:
+  // ipv4 - IPv4 addresses only (Default)
+  // ipv6 - IPv6 addresses only
+  // +kubebuilder:validation:Optional
+  // +kubebuilder:validation:MaxItems=2
+  IPFamilyPriority []string `json:"ipFamily,omitempty"`
+}
+
+// VSpherePlatformPlacementConstraint is the context information for VM placements within a failure domain
+type VSpherePlatformPlacementConstraint struct {
+  // resourcePool is the absolute path of the resource pool where virtual machines will be
+  // created. The absolute path is of the form /<datacenter>/host/<cluster>/Resources/<resourcepool>.
```

Comment:
Perhaps we can add a regex for this? Or is that too complicated?
Wondering if there's a set of characters allowed for this/more a set that aren't allowed